app/util/vec2.ts

```diff
@@ -0,0 +1,53 @@
+import { Vector2 } from '../../vendor/threejs/vector2';
+
+type TVecConstructorArgs = [number, number] | [IVec2] | [];
+
+/**
+ * Math for 2d vectors
+ * Use a THREE.Vector2 class as a base class
+ * @see https://threejs.org/docs/#api/en/math/Vector2
+ */
+export class Vec2 extends Vector2 {
```

Comment:
I suggest using a Vector class for the geometry Math. In the ideal world it's good to have some library with classes like `Rect`, `Line`, `Vec`. Unfortunately, I didn't find any good and lightweight library for JS. So I just added a Vector2 class from the THREE.js lib, cuz it doesn't have any dependencies and well known in the JS world.